src/spec.rs

```diff
@@ -273,3 +277,247 @@ pub(crate) fn path_to_string<T: AsRef<Path>>(path: T) -> Box<str> {
         maybe_original.unwrap().into()
     }
 }
+
+#[cfg(feature = "tokio-experimental")]
+pub struct AsyncCentralDirectoryEnd {
+    pub disk_number: u16,
+    pub disk_with_central_directory: u16,
+    pub number_of_files_on_this_disk: u16,
+    pub number_of_files: u16,
+    pub central_directory_size: u32,
+    pub central_directory_offset: u32,
+    pub zip_file_comment: Arc<[u8]>,
```

Comment:
Don't create a new field for this, just change the `zip_file_comment` in the existing `CentralDirectoryEnd` to be either an `Option<Arc<[u8]>>` (it'll usually be empty) or a generic type parameter.